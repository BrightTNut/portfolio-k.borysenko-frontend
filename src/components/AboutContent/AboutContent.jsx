import { toast } from 'react-toastify';
import { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import Paper from '../common/Paper/Paper';
import { ArrowIcon } from '../icons';
import img from '../../images/about.png';
import s from './AboutContent.module.css';
import LightButton from '../../uikit/LightButton/LightButton';

const AboutContent = () => {
  const navigate = useNavigate();

  const handleDownloadBtnClick = useCallback(() => {
    fetch('/cv_frontend_react_borysenko.pdf')
      .then(response => response.blob())
      .then(blob => {
        const url = window.URL.createObjectURL(blob);
        const a = document.createElement('a');
        a.href = url;
        a.download = 'cv_frontend_react_borysenko.pdf';
        document.body.appendChild(a);
        a.click();
        a.remove();
        window.URL.revokeObjectURL(url);
        toast.success('CV was downloaded successfully');
      })
      .catch(error => {
        toast.error(
          `There was an error downloading the file: ${error.message}`,
        );
      });
  }, []);

  const handleCertificateBtnClick = () => {
    navigate('/certificates');
  };

  return (
    <Paper>
      <div className={s.container}>
        <div className={s.top_container}>
          <div className={s.btn_container}>
            <div className={s.position}>Fullstack Developer</div>
            <LightButton name="Download CV" onClick={handleDownloadBtnClick} />
            <LightButton
              name="Certificates"
              onClick={handleCertificateBtnClick}
            />
          </div>
          <ArrowIcon className={s.arrow_icon} />

          <h1 className={s.title}>Kateryna Borysenko</h1>
        </div>
        <div className={s.bottom_container}>
          <section className={s.description_section}>
            <p className={s.paragraph}>
              <span className={s.accent}>Hey, that's my name up there. </span>I
              am a Fullstack Developer with a rich two-year experience in
              creating dynamic web applications. Originally from the picturesque
              regions of Eastern Ukraine, I have recently found a new home in
              sunny Florida â€“ a change brought about by turbulent times in my
              homeland.
            </p>
            <p className={s.paragraph}>
              My professional story is a path of relentless discoveries and bold
              technological solutions. I have mastered a multitude of tools from
              HTML5 to Node.js and take pride in transforming code into art that
              enhances people's lives.
            </p>
            <p className={s.paragraph}>
              I believe that every project is not just a task, but a story that
              requires a personalized approach and creativity. My attention to
              detail, ability to see the bigger picture, and passion for
              innovation allow me to create products that are not only
              functional but also inspiring.
            </p>
            <p className={s.paragraph}>
              <span className={s.accent}>
                Today, actively seeking new opportunities, I am ready to apply
                my skills to ambitious projects.
              </span>
            </p>
          </section>
          <section className={s.quote_section}>
            <img className={s.image} src={img} alt="Developer" />
          </section>
          <section className={s.experience_section}>
            <p className={s.paragraph}>
              <span className={s.accent}>Proficient in </span>
              React, Redux, Redux thunk, React testing library, HTML5, CSS3,
              SASS, JavaScript, TypeScript.
            </p>
            <p className={s.paragraph}>
              <span className={s.accent}>Familiar with </span>
              NodeJS, Express, Mongo DB, React Native, Next JS.
            </p>
            <p className={s.paragraph}>
              <span className={s.accent}>Other Tools: </span>
              Figma, Git, GitHub, GitLab, Jira, Trello.
            </p>
            <p className={s.paragraph}>
              <span className={s.accent}>Methodologies: </span>
              Agile, Scrum, Responsive and Adaptive Design, BEM.
            </p>
          </section>
        </div>
      </div>
    </Paper>
  );
};

export default AboutContent;
