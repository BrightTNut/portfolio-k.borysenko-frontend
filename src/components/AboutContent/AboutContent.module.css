.container {
  max-width: 1020px;
}

.top_container {
  display: flex;
}

.title_container {
  width: 75%;
}

.btn_container {
  display: flex;
  flex-direction: column;
  width: 25%;
  gap: 20px;
  align-items: flex-end;
  margin-bottom: 20px;
}

.position {
  font-family: var(--font-heading);
  letter-spacing: 1px;
}

.title {
  font-family: var(--font-heading);
  font-size: var(--font-size-h2);
  margin: 25px 0;
}

.arrow_icon {
  width: 40px;
}

.bottom_container {
  position: relative;
  display: flex;
  align-items: flex-end;
  gap: 40px;
}

.description_section {
  width: 50%;
}

.image_section {
  background: linear-gradient(45deg, #808080, #cc6896);
  border: 4px solid var(--bg-primary);
  box-sizing: border-box;
  height: 430px;
  width: 300px;
}

.image {
  position: absolute;
  height: 490px;
  bottom: 4px;
  left: 300px;
}

.experience_section {
  width: 20%;
}

.accent {
  font-weight: var(--font-weight-bold);
}

.paragraph:not(:last-child) {
  margin-bottom: 24px;
}

/* @media screen and (max-height: 1100px), screen and (max-width: 1160px) {
  .container {
    max-width: 850px;
  }

  .title {
    margin: 30px 0;
  }

  .bottom_container {
    flex-wrap: wrap;
    align-items: start;
  }
  .description_section {
    width: 100%;
  }
  .image {
    left: -200px;
  }
  .experience_section {
    width: 40%;
  }
}

@media screen and (max-width: 900px) {
} */
