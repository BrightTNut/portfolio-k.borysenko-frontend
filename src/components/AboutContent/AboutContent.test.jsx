import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import AboutContent from './AboutContent';
import { toast } from 'react-toastify';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn().mockImplementation(() => () => {}),
}));

jest.mock('react-toastify', () => ({
  toast: {
    success: jest.fn(),
    error: jest.fn(),
  },
}));

describe('AboutContent', () => {
  it('navigates to the certificates page on button click', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    fireEvent.click(screen.getByText('Certificates'));
  });

  it('handles errors during CV download', async () => {
    global.fetch = jest.fn(() => Promise.reject(new Error('Fetch error')));
    render(<AboutContent />, { wrapper: BrowserRouter });
    fireEvent.click(screen.getByText('Download CV'));
    await waitFor(() =>
      expect(toast.error).toHaveBeenCalledWith(
        'There was an error downloading the file: Fetch error',
      ),
    );
  });

  it('displays the fullstack developer position', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    expect(screen.getByText('Fullstack Developer')).toBeInTheDocument();
  });

  it('displays the about section with personal description', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    expect(
      screen.getByText("Hey, that's my name up there."),
    ).toBeInTheDocument();
  });

  it('displays the professional skills section', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    expect(screen.getByText('Proficient in')).toBeInTheDocument();
    expect(screen.getByText('Familiar with')).toBeInTheDocument();
    expect(screen.getByText('Other Tools:')).toBeInTheDocument();
    expect(screen.getByText('Methodologies:')).toBeInTheDocument();
  });

  it('displays the download CV and certificates buttons', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    expect(screen.getByText('Download CV')).toBeInTheDocument();
    expect(screen.getByText('Certificates')).toBeInTheDocument();
  });

  it('displays the image in the quote section', () => {
    render(<AboutContent />, { wrapper: BrowserRouter });
    expect(screen.getByAltText('Developer')).toBeInTheDocument();
  });

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<AboutContent />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
