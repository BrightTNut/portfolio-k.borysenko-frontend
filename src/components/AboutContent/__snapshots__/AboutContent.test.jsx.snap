// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutContent renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="top_container"
      >
        <div
          class="title_container"
        >
          <div
            class="position"
          >
            Fullstack Developer
          </div>
          <h2
            class="title"
          >
            Kateryna Borysenko
          </h2>
        </div>
        <div
          class="btn_container"
          style="opacity: 0; transform: translateX(-300px) translateY(50px) translateZ(0);"
        >
          <button
            class="button"
          >
            Download CV
          </button>
          <button
            class="button"
          >
            Certificates
          </button>
        </div>
      </div>
      <div
        class="bottom_container"
      >
        <section
          class="description_section"
        >
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Hey, that's my name up there.
              <svg
                class="arrow_icon"
              >
                arrow.svg
              </svg>
            </span>
            I'm a Fullstack Developer with a rich two-year experience in creating dynamic web applications. Originally from the picturesque regions of Eastern Ukraine, I have recently found a new home in sunny Florida â€“ a change brought about by turbulent times in my homeland.
          </p>
          <p
            class="paragraph"
          >
            My professional story is a path of relentless discoveries and bold technological solutions. I have mastered a multitude of tools from HTML5 to Node.js and take pride in transforming code into art that enhances people's lives.
          </p>
          <p
            class="paragraph"
          >
            I believe that every project is not just a task, but a story that requires a personalized approach and creativity. My attention to detail, ability to see the bigger picture, and passion for innovation allow me to create products that are not only functional but also inspiring.
          </p>
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Today, actively seeking new opportunities, I am ready to apply my skills to ambitious projects.
            </span>
          </p>
        </section>
        <section
          class="quote_section"
        >
          <img
            alt="Developer"
            class="image"
            loading="lazy"
            src="about.png"
          />
        </section>
        <section
          class="experience_section"
        >
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Proficient in 
            </span>
            React, Redux, Redux thunk, React testing library, HTML5, CSS3, SASS, JavaScript, TypeScript, REST API.
          </p>
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Familiar with 
            </span>
            NodeJS, Express, Mongo DB, React Native, Next JS, GraphQL, Apollo.
          </p>
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Other Tools: 
            </span>
            Figma, Git, GitHub, GitLab, Jira, Trello.
          </p>
          <p
            class="paragraph"
          >
            <span
              class="accent"
            >
              Methodologies: 
            </span>
            Agile, Scrum, Responsive and Adaptive Design, BEM.
          </p>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
